crates/shared/src/price_estimation/baseline.rs

```diff
@@ -304,15 +301,14 @@ impl BaselinePriceEstimator {
         O: Ord,
     {
         debug_assert!(sell_token != buy_token);
-        debug_assert!(!amount.is_zero());
 
         let path_candidates = self.base_tokens.path_candidates(sell_token, buy_token);
         let best_path = path_candidates
             .iter()
-            .max_by_key(|path| comparison(amount, path, pools))
+            .max_by_key(|path| comparison(amount.get(), path, pools))
```

Comment:
You could also derive `Ord` and `PartialOrd` for `NonZeroU256`.

Reply:
It doesn't allow to change this line however, does it?

Reply:
No, it does not. Didn't look closely enough. Just saw `.get()` in a `comparison()` function call which made me think you had to get at the inner type to compare it.